cmake/externals/sentencepieceproject.cmake

```diff
@@ -1,21 +1,22 @@
+# spm is abbreviation of sentencepiece to meet the path length limits on Windows
+
+set(spm_patches
+    # use protobuf provided by ORT
+    "${PROJECT_SOURCE_DIR}/cmake/externals/sentencepieceproject.protobuf.patch"
```

Comment:
How do you create another patch package based on an existing one? To be simple, should we just create one since the changes for iOS can be enclosed by if (iOS)?

Reply:
the patch is actually based on the original repo. but both patches can apply cleanly together.

actually, realized that the protobuf-related patch only applies for the embedded in ORT build, so keeping that separate

Reply:
It's kind of special that you can apply two patches together. 
If I also want to do a patch for spm for Windows build, do I need to do patches locally first or directly make the change? If latter, how can I avoid the conflict. Overall, these multiple patches method add unnecessary complex for the process. And I don't see any benefits from the multiple patches. So, let's keep the process simple and straightforward.

Reply:
It's ok to keep them be separated in this PR, I will have a PR to make the combined build and standalone build with same configuration. Then, they can be merged.

Reply:
ok, we can merge them later. it doesn't hurt to always include this ios fix patch.